### PR TITLE
docs, tests, fix: fit fibonacci_fast to contributing guidelines

### PR DESCRIPTION
**Pull request by setbit123**
#### Description of Change
Modify the fibonacci_fast.cpp to fit the project style requirements.
Solve a small part of issue #2456 

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by setbit123** [comment]
cc @Panquesito7 @realstealthninja

File: math/fibonacci_fast.cpp [hunk 1]
```diff
@@ -1,36 +1,41 @@
...
 /**
- * maximum number that can be computed - The result after 93 cannot be stored
- * in a `uint64_t` data type.
+ * @brief Maximum Fibonacci number that can be computed
+ *
+ * @details
+ * The result after 93 cannot be stored in a `uint64_t` data type.
  */
-
 #define MAX 93
```
Comment (realstealthninja) [inline math/fibonacci_fast.cpp:28]: should use a constexpr or const instead of macros

File: math/fibonacci_fast.cpp [hunk 2]
```diff
@@ -39,19 +44,120 @@ uint64_t fib(uint64_t n) {
             << "Cannot compute for n>93 due to limit of 64-bit integers\n";
         return 0;
```
Comment (realstealthninja) [inline math/fibonacci_fast.cpp:45]: instead of output to cerr, throw this as an invalid_argument error instead.

File: math/fibonacci_fast.cpp [hunk 3]
```diff
@@ -39,19 +44,120 @@ uint64_t fib(uint64_t n) {
...
+    assert(fib(83) == 99194853094755497);
+    assert(fib(84) == 160500643816367088);
+    assert(fib(85) == 259695496911122585);
+    assert(fib(86) == 420196140727489673);
+    assert(fib(87) == 679891637638612258);
+    assert(fib(88) == 1100087778366101931);
+    assert(fib(89) == 1779979416004714189);
+    assert(fib(90) == 2880067194370816120);
+    assert(fib(91) == 4660046610375530309);
+    assert(fib(92) == 7540113804746346429);
```
Comment (realstealthninja) [inline math/fibonacci_fast.cpp:152]: catch the above mentioned error and assert that it's message matches.

**Review by realstealthninja — CHANGES_REQUESTED** [review]
Thank you for your contribution!

File: math/fibonacci_fast.cpp (hunk 1 above)
Comment (setbit123) [reply math/fibonacci_fast.cpp]: I have already revised it to the following content:

constexpr uint64_t MAX = 93;

File: math/fibonacci_fast.cpp (hunk 2 above)
Comment (setbit123) [reply math/fibonacci_fast.cpp:46]: I have already revised it to the following content:

    if (n >= MAX) {
        throw std::invalid_argument("Cannot compute for n>=" + std::to_string(MAX) +
                                    " due to limit of 64-bit integers");
        return 0;
    }

File: math/fibonacci_fast.cpp (hunk 3 above)
Comment (setbit123) [reply math/fibonacci_fast.cpp:154]: I have already revised it to the following content:

// Test for invalid Fibonacci numbers
    try {
        fib(MAX + 1);
        assert(false && "Expected an invalid_argument exception to be thrown");
    } catch (const std::invalid_argument& e) {
        const std::string expected_message = "Cannot compute for n>=" + std::to_string(MAX) +
                                             " due to limit of 64-bit integers";
        assert(e.what() == expected_message);
    }

**Comment by setbit123** [comment]
Thank you very much for providing valuable suggestions. I have made an attempt to revise my PR according to your advice.

cc @realstealthninja

**Review by realstealthninja — APPROVED** [review]
Very nicely done!

**Comment by setbit123** [comment]
Thank you for your response! I noticed the following errors in Compile checks (windows-latest):
error C2039: 'to_string': is not a member of 'std'
error C3861: 'to_string': identifier not found
error C2512: 'std::invalid_argument': no appropriate default constructor available
I tried adding some header files to solve this issue. Looking forward to your reply!

cc @realstealthninja

**Comment by setbit123** [comment]
Thank you very much for taking the time to reply to me despite your busy schedule!
I noticed the following errors in Compile checks (windows-latest):
error C2039: ‘to_string’: is not a member of ‘std’
error C3861: ‘to_string’: identifier not found
error C2512: ‘std::invalid_argument’: no appropriate default constructor available
I tried adding some header files to solve this issue. Looking forward to your reply!



https://github.com/TheAlgorithms/C-Plus-Plus/pull/2893
&nbsp;


史恩僮
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年11月23日(星期六) 中午11:27
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [TheAlgorithms/C-Plus-Plus] docs, tests, fix: fit fibonacci_fast to contributing guidelines (PR #2893)














 
@realstealthninja approved this pull request.
 
Very nicely done!

**Comment by setbit123** [comment]
It seems there is a new code style check that wasn’t present before. I made the necessary modifications.

cc @realstealthninja

**Comment by setbit123** [comment]
According to the check results, it seems there is still an issue with token permissions. I'm not sure if I have resolved this issue.

cc @realstealthninja

**Comment by realstealthninja** [comment]
Sorry for that, I'm working on finding a fix

**Comment by setbit123** [comment]
Looking forward to your review after you finish your work! If my PR doesn’t pass the testing, I might have to bother you again for further modifications.

cc @realstealthninja

**Comment by setbit123** [comment]
Additionally, I found that one test consistently reports an error, but I don’t know how to resolve it. If you could provide assistance, I would greatly appreciate it.

https://github.com/setbit123/C-Plus-Plus/actions/runs/11996941748

cc @realstealthninja

**Comment by realstealthninja** [comment]
> Additionally, I found that one test consistently reports an error, but I don’t know how to resolve it. If you could provide assistance, I would greatly appreciate it.
> 
> https://github.com/setbit123/C-Plus-Plus/actions/runs/11996941748
> 
> cc @realstealthninja 

Its just GitHub pages nothing to worry about. I'm just going to merge this

**Comment by setbit123** [comment]
Thank you for your reply!

**Pull request merged**